control-plane/api-gateway/gatekeeper/deployment.go

```diff
@@ -69,13 +72,12 @@ func (g *Gatekeeper) deleteDeployment(ctx context.Context) error {
 func (g *Gatekeeper) deployment() *appsv1.Deployment {
 	return &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        g.Gateway.Name,
-			Namespace:   g.Gateway.Namespace,
-			Labels:      apigateway.LabelsForGateway(&g.Gateway),
-			Annotations: g.HelmConfig.CopyAnnotations,
```

Comment:
Can you just add a small TODO that we need to add this back in with the whitelisted values from the class config? We can handle it in a follow-up.

You're right that this shouldn't be coming from Helm, but I just want to make sure we don't accidentally drop it completely.